x/shield/keeper/pool.go

```diff
@@ -222,23 +222,30 @@ func (k Keeper) ValidatePoolDuration(ctx sdk.Context, timeDuration, numBlocks in
 func (k Keeper) WithdrawFromPools(ctx sdk.Context, addr sdk.AccAddress, amount sdk.Coins) {
 	provider, _ := k.GetProvider(ctx, addr)
 	withdrawAmtDec := sdk.NewDecFromInt(amount.AmountOf(k.sk.BondDenom(ctx)))
-	collateralDec := sdk.NewDecFromInt(provider.Collateral.AmountOf(k.sk.BondDenom(ctx)))
-	proportion := withdrawAmtDec.Quo(collateralDec)
-	if amount.IsAnyGT(provider.Collateral) {
+	withdrawableAmtDec := sdk.NewDecFromInt(provider.Collateral.AmountOf(k.sk.BondDenom(ctx)).Sub(provider.Withdrawal))
+	proportion := withdrawAmtDec.Quo(withdrawableAmtDec)
+	if amount.AmountOf(k.sk.BondDenom(ctx)).ToDec().GT(withdrawableAmtDec) {
```

Comment:
Ah sorry forgot to adjust this part after adding the withdrawal field .. Technically this shouldn't happen, right?